### PR TITLE
Handle blob: urls

### PR DESCRIPTION
**Pull request by stereobooster**
Fix for https://github.com/peterbe/minimalcss/issues/17 There is example file for test

### CODE REVIEW

File: src/run.js [hunk 1]
```diff
@@ -191,9 +190,20 @@ const minimalcss = async options => {
...
+              const response = await fetch(stylesheet.href)
+              const text = await response.text()
+              const ast = csstree.parse(text, {
+                parseValue: false,
+                parseRulePrelude: false
+              })
+              stylesheetAstObjects[url] = csstree.toPlainObject(ast)
+              stylesheetContents[url] = text
+            } else {
+              throw new Error(`${stylesheet.href} not in stylesheetAstObjects!`)
```
Comment (stereobooster) [inline src/run.js]: much simpler solution would be ignore `blob:` urls

File: src/run.js [hunk 2]
```diff
@@ -191,9 +190,20 @@ const minimalcss = async options => {
...
           )
         })
-        .forEach(stylesheet => {
+        .map(async stylesheet => {
           if (!stylesheetAstObjects[stylesheet.href]) {
-            throw new Error(`${stylesheet.href} not in stylesheetAstObjects!`)
+            if (stylesheet.href.startsWith('blob:')) {
+              const response = await fetch(stylesheet.href)
+              const text = await response.text()
+              const ast = csstree.parse(text, {
```
Comment (stereobooster) [inline src/run.js]: `csstree` is not defined in this context :/

File: src/run.js [hunk 3]
```diff
@@ -48,18 +48,17 @@ const minimalcss = async options => {
...
-            parseRulePrelude: false
-          })
-          stylesheetAstObjects[url] = csstree.toPlainObject(ast)
-          stylesheetContents[url] = text
+        const text = await response.text()
+        const ast = csstree.parse(text, {
+          parseValue: false,
+          parseRulePrelude: false
         })
+        stylesheetAstObjects[url] = csstree.toPlainObject(ast)
```
Comment (stereobooster) [inline src/run.js]: Or

```js
    if (url.startsWith('blob:')) {
          page.evaluate(url => {
            return fetch(url).then(response => response.text())
          }, url).then(text => {
            console.log(text)
            const ast = csstree.parse(text, {
              parseValue: false,
              parseRulePrelude: false
            })
            stylesheetAstObjects[url] = csstree.toPlainObject(ast)
            stylesheetContents[url] = text
          })
        } else {
          response.text().then(text => {
            const ast = csstree.parse(text, {
              parseValue: false,
              parseRulePrelude: false
            })
            stylesheetAstObjects[url] = csstree.toPlainObject(ast)
            stylesheetContents[url] = text
          })
        }
```

File: src/run.js [hunk 4]
```diff
@@ -193,7 +193,12 @@ const minimalcss = async options => {
         })
         .forEach(stylesheet => {
           if (!stylesheetAstObjects[stylesheet.href]) {
-            throw new Error(`${stylesheet.href} not in stylesheetAstObjects!`)
+            if (stylesheet.href.startsWith('blob:')) {
+              // TODO: do we need to do something here?
```
Comment (peterbe) [inline src/run.js]: I think this is the right thing to do. It's sad but it's what it is. 

It's not that different from the fact that `minimalcss` doesn't clean up any `<style>` tags on the page. (Perhaps it should in a future version. But that's a story for another day) 
If someone uses a blob to inject a bunch of "fat" CSS then I think it's best to just ignore it. 

However, I would like one change. Can you remove this if statement and weave it into the filter loop a couple of lines above?
Perhaps something like this:
```javascript
.filter(link => {
    return (
       link.href && !link.href.startsWith('blob:') &&
       ...
    )
})
```
What do you think about that?

**Comment by stereobooster** [comment]
Done

**Comment by peterbe** [comment]
Great work! I think this deserves a version bump.

**Comment by peterbe** [comment]
0.2.3 out.

**Pull request merged**